src/shape.py

```diff
@@ -76,3 +76,65 @@ def create_shape(type: str, start_row: int, start_col: int) -> Shape:
         return SquareShape(start_row, start_col)
 
     raise NotImplementedError("Shape type not implemented")
+
+
+"""
+    U - - 
+    U - - 
+    U U -
+
+    U U U
+    U - -
+    - - -
+
+    - U U
+    - - U
+    - - U
+
+    - - -
+    - - U
+    U U U
+    """
+
+
+def rotate(shape: Shape) -> Shape:
+    unit_row_array = []
+    unit_col_array = []
+
+    for unit in shape.units:
+        unit_row_array.append(unit.row)
+        unit_col_array.append(unit.col)
+
+    row_diff = max(unit_row_array) - min(unit_row_array) + 1
+    col_diff = max(unit_col_array) - min(unit_col_array) + 1
+    min_value = min(min(unit_row_array), min(unit_col_array))
+    dimension = max(row_diff, col_diff)
+
+    grid = [[None for _ in range(dimension)] for _ in range(dimension)]
+    second_grid = [[None for _ in range(dimension)] for _ in range(dimension)]
+
+    for unit in shape.units:
+        grid[unit.row-min_value][unit.col-min_value] = unit
+
+    for row in range(len(grid)):
+        for col in range(len(grid[row])):
+            second_grid[col][len(grid)-1-row] = grid[row][col]
+
+    """
+            second[0][2]   grid[0][0]
+            second[1][2]   grid[0][1]
+            second[2][2]   grid[0][2]
+
+            second[0][1]   grid[1][0]
+            second[1][1]   grid[1][1]
+            second[2][0]   grid[1][2]
+    """
+
+    new_shape = Shape(0, 0)
+
+    for row in range(len(second_grid)):
+        for col in range(len(second_grid[row])):
+            if second_grid[row][col] != None:
```

Comment:
None icin `!=` gibi mantiksal operatorler yerine direkt `is not None` veya direkt `if second_grid[row][col]` yazabiliriz cunku bu ifade `None` degilse devam etmek istiyorsun. Mesela `None` oldugunda devam etmek isteseydin `if not second_grid[row][col]` yazilabilirdi.

```suggestion
            if second_grid[row][col] is not None:
```